src/hotspot/cpu/riscv/vm_version_riscv.cpp

```diff
@@ -240,9 +240,15 @@ void VM_Version::common_initialize() {
   }
 
   // UseZvfh (depends on RVV)
-  if (UseZvfh && !UseRVV) {
-    warning("Cannot enable UseZvfh on cpu without RVV support.");
-    FLAG_SET_DEFAULT(UseZvfh, false);
+  if (UseZvfh) {
+    if (!UseRVV) {
+      warning("Cannot enable UseZvfh on cpu without RVV support.");
+      FLAG_SET_DEFAULT(UseZvfh, false);
+    }
+    if (!UseZfh) {
+      warning("Cannot enable UseZvfh on cpu without Zfh support.");
+      FLAG_SET_DEFAULT(UseZvfh, false);
+    }
```

Comment:
For QEMU, Zvfh also means Zvfhmin and Zfhmin. So there is no Zfh if you only enable Zvfh when start QEMU.
That means hwprobe syscall will detects Zvfh and Zfhmin but no Zfh. Is there a similar issue for this case?

Reply:
One question: Not sure if I understand you correctly, seems to me Zfh is also detected and automatically set by hwprobe when running with qemu?

Reply:
For this pr, it tries to address the situation: when neither Zfh/Zvfh can not be enabled automatically (no matter what's the reason), so user could enable Zvfh manually to get some optimization, in that case it will fail to get such optimization even if with -XX:+PrintFlagsFinal it shows that UseZvfh == true, this is very confusing.
So to make this situation more understanable for user (maybe more for developer) of jdk, we'd better to automatically unset UseZvfh when UseZfh is false.